lib/LedgerSMB/Upgrade_Tests.pm

```diff
@@ -214,13 +231,37 @@ Tooltip for each button
 =cut
 
 has tooltips => (is => 'ro',
-    isa => 'HashRef[Str]',
-    default => sub {
-        return {
-            'Save and Retry' => marktext('Save the fixes provided and attempt to continue migration'),
-            'Cancel' => marktext('Cancel the <b>migration</b>')
-    }},
-    required => 0);
+    isa => 'Maybe[HashRef[Str]]', required => 0,
+    default => undef,   # Force initializer call
+    initializer => sub {
+        my ( $self, $value, $writer_sub_ref, $attribute_meta ) = @_;
+        for ($value) {
+            die LedgerSMB::I18N::text(q(No button '[_1]' in test '[_2]'),$_,$self->{name})
+                if not grep( /^$_$/, @{$self->{buttons}});
+        }
+        my %defaults = ('Save and Retry' => marktext('Save the fixes provided and attempt to continue migration'),
+                                'Cancel' => marktext('Cancel the <b>whole migration</b>'));
+        for (keys %defaults) {
+            $value->{$_} = $defaults{$_}
+                if not $value->{$_};
+        }
+        $writer_sub_ref->($value);
+    }
+);
```

Comment:
The initializer feels over-engineered. How many more confirmations are we going to be seeing? Can we simply go for listing the confirmations that make sense for a specific test, deferring rendering of the buttons to the test-driver, supporting save(and retry), cancel and skip for now?